packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -696,6 +696,12 @@ class TextSelectionOverlay {
   // corresponds to, in global coordinates.
   late double _endHandleDragTarget;
 
+  // The edge to hold in place when the selection is inverted.
+  int? _oppositeEdge;
```

Comment:
Nit: Maybe pivotPoint or something? Not sure if that's any more understandable, up to you...

Reply:
Or rather than being an integer, should it just indicate whether the start or end handle is being dragged? I feel like the indicator contains information that's duplicated in _selection.

Reply:
Thinking about this some more I think you're right. I think instead of having `_oppositeEdge` and `_dragStartSelectionIsCollapsed`, I can just store `_dragStartSelection` and use it directly. I think we know what handle is being dragged depending on what method we are in `_handleSelectionEndHandleDragUpdate` and `_handleSelectionStartHandleDragUpdate`.

Thinking out loud about some of the behaviors:

End handle:
When the selection is normalized and we drag the end handle, for the duration of the drag we can hold the baseOffset in place at the base.
When the selection is inverted and we drag the end handle, for the duration of the drag we can hold the extentOffset in place at the base.

Start handle:
When the selection is normalized and we drag the start handle, for the duration of the drag we can hold the extentOffset in place at the base.
When the selection is inverted and we drag the start handle, for the duration of the drag we can hold the baseOffset in place at the base.

Concrete example:
selection format: (baseOffset, extentOffset)
initial selection (end handle at position 10): (5,10)
drag end handle forward (to position 15): (5, 15)
drag end handle backwards & invert (to position 2) - release drag: (5, 2)
new end handle drag forward (to position 8) (end handle is on position 5 and iOS always sets the new offset as the extentOffset) : (2, 8) since the start handle is on position 2, so that position should be held in place while dragging the end handle.